test/test_intensify/test_successive_halving.py

```diff
@@ -0,0 +1,240 @@
+import unittest
+from nose.plugins.attrib import attr
+
+import logging
+import numpy as np
+import time
+
+from ConfigSpace import Configuration, ConfigurationSpace
+from ConfigSpace.hyperparameters import UniformIntegerHyperparameter
+
+from smac.scenario.scenario import Scenario
+from smac.tae.execute_func import ExecuteTAFuncDict
+from smac.intensification.successive_halving import SuccessiveHalving
+from smac.runhistory.runhistory import RunHistory, RunKey
+from smac.optimizer.objective import average_cost, sum_cost
+from smac.tae.execute_ta_run import StatusType
+from smac.stats.stats import Stats
+from smac.utils.io.traj_logging import TrajLogger
+
+
+def get_config_space():
+    cs = ConfigurationSpace()
+    cs.add_hyperparameter(UniformIntegerHyperparameter(name='a',
+                                                       lower=0,
+                                                       upper=100))
+    cs.add_hyperparameter(UniformIntegerHyperparameter(name='b',
+                                                       lower=0,
+                                                       upper=100))
+    return cs
+
+
+class TestSuccessiveHalving(unittest.TestCase):
+
+    def setUp(self):
+        unittest.TestCase.setUp(self)
+
+        self.rh = RunHistory(aggregate_func=average_cost)
+        self.cs = get_config_space()
+        self.config1 = Configuration(self.cs,
+                                     values={'a': 0, 'b': 100})
+        self.config2 = Configuration(self.cs,
+                                     values={'a': 100, 'b': 0})
+        self.config3 = Configuration(self.cs,
+                                     values={'a': 100, 'b': 100})
+
+        self.scen = Scenario({"cutoff_time": 2, 'cs': self.cs,
+                              "run_obj": 'runtime',
+                              "output_dir": ''})
+        self.stats = Stats(scenario=self.scen)
+        self.stats.start_timing()
+
+        self.logger = logging.getLogger(self.__module__ + "." + self.__class__.__name__)
+
+    def test_top_k(self):
```

Comment:
I think this test needs to be updated. In case of instances, one needs to make sure that all configurations are compared on the same instances, and the instances must be only the ones which are so far chosen by successive halving. The current test setting should therefore actually fail.